### DISCUSSION TITLE
Libint update

### DISCUSSION THREAD

**Discussion by pseewald** [cp2k/cp2k#2587] [Team Posts]
This is an announcement that I will update the libint version used in CP2K on **Monday, 17th of June** (if there are no objections). Due to major API changes, libint 1 will no longer be supported, only libint version 2.5.0 and above will be compatible with CP2K. All regression testers running cp2k with libint need to be updated.

Please have a look at the pull request cp2k/cp2k#300 for further instructions on how to build libint & CP2K.

Some additional information about libint 2:
- Libint 2 is a compiler that generates custom libraries. Due to the additional dependencies and the complicated build procedure, we provide automatic releases of generated libint 2 libraries on [cp2k/libint-cp2k](https://github.com/cp2k/libint-cp2k) for convenience that have no additional dependencies (thanks to @dev-zero for setting up the CI).
- Libint 2 supports 3-center and 2-center ERIs and other integral types. Developers who like to know more about features of libint and how to use it should read the programmer's manual that comes with one of the generated libraries (run `./configure` and `make doc` to generate the manual).

**Comment by oschuett** [comment]
It seems Libint2 does not build with GCC 4.8.5 :
```
root@46416f7a87d5:/opt/cp2k-toolchain/build/libint-v2.5.0-cp2k-lmax-4# make
CXX src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.lo
In file included from ./include/libint2.h:34:0,
                 from src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.cc:21:
./include/libint2/util/generated/libint2_params.h:29:22: error: missing binary operator before token "("
 #    if __has_include(<libint2_params.h>)
                      ^
In file included from ./include/libint2.h:35:0,
                 from src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.cc:21:
./include/libint2/util/generated/libint2_types.h:28:19: error: missing binary operator before token "("
 # if __has_include(<libint2_types.h>)
                   ^
In file included from ./include/libint2.h:43:0,
                 from src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.cc:21:
./include/libint2/util/generated/libint2_iface.h:28:19: error: missing binary operator before token "("
 # if __has_include(<libint2_iface.h>)
                   ^
src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.cc:26:85: error: 'Libint_t' does not name a type
 void CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0(const Libint_t* inteval, LIBINT2_REALTYPE* target, const LIBINT2_REALTYPE* src0, const LIBINT2_REALTYPE* src1) {
                                                                                     ^
src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.cc:26:95: error: ISO C++ forbids declaration of 'inteval' with no type [-fpermissive]
 void CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0(const Libint_t* inteval, LIBINT2_REALTYPE* target, const LIBINT2_REALTYPE* src0, const LIBINT2_REALTYPE* src1) {
                                                                                               ^
MakeSuffixRules:15: recipe for target 'src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.lo' failed
make: *** [src/CR_DerivGaussP1InBra_aB_H__0__G100__1___TwoPRep_unit__0__G__1___Ab__up_0.lo] Error 1
```
Since we're anyways planning to drop support for GCC4.8 due to https://github.com/cp2k/cp2k/pull/356, I'd say we don't bother.

**Comment by alazzaro** [comment]
It seems that it is missing the `-std=c++11`, not sure if this is the case...

**Comment by rathann** [comment]
For what it's worth, on RHEL 7 and derivatives, one can use DevToolset SCL. Version 7 comes with gcc 7.3.1 and version 8 comes with 8.2.1, both of which should be good enough to build C++11 code correctly. So, if you're worried about compiling with RHEL 7's gcc-4.8.5, there is an easy workaround.

http://mirror.centos.org/centos/7/sclo/x86_64/rh/devtoolset-7/
http://mirror.centos.org/centos/7/sclo/x86_64/rh/devtoolset-8/
